### PR TITLE
Remove the need for a separate branch in integration tests + cleanup

### PR DESCRIPTION
**Pull request by baldimir**
- Updates the dependencies versions. 
- Removes the need to have a separate branch to run integration tests correctly. 
- Adds proper README.

### CODE REVIEW

**Comment by baldimir** [comment]
@hellowdan please reassign your review request if needed.

File: README.md [hunk 1]
```diff
@@ -1,3 +1,93 @@
...
+  <activation>
+    <property>
+      <name>unpackbuild.version</name>
+    </property>
+  </activation>
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
```
Comment (jstastny-cz) [inline README.md]: Wouldn't it be more convenient to use the user property instead? Would definition of `maven.main.skip` property work the same? https://maven.apache.org/plugins/maven-compiler-plugin/compile-mojo.html#skipMain
```
<properties>
  <maven.main.skip>true</maven.main.skip>
</properties>
```
Or at least add a note that the compiler config can be replaced by the property?

**Review by jstastny-cz — COMMENTED** [review]
LGTM, just a minor comment.

File: README.md (hunk 1 above)
Comment (baldimir) [reply README.md]: Fixed.

**Review by hellowdan — APPROVED** [review]

**Comment by hellowdan** [comment]
I'm waiting for @MarianMacik review before merging it.

File: unpack-build-maven-plugin-itests/unpack-build-maven-plugin/pom.xml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <parent>
+    <groupId>org.kie</groupId>
+    <artifactId>unpack-build-maven-plugin-itests</artifactId>
+    <version>1.0-SNAPSHOT</version>
+  </parent>
+
+  <modelVersion>4.0.0</modelVersion>
+  <artifactId>unpack-build-maven-plugin</artifactId>
```
Comment (MarianMacik) [inline unpack-build-maven-plugin-itests/unpack-build-maven-plugin/pom.xml:12]: Maybe add here an explanation why this has the same GAV as the "real" unpack-build-maven-plugin. Another option would be to rename this to `-dummy` and build it before the `tests` module and set up a clean plugin execution to clean it afterwards. Then the `tests` module would run.

File: unpack-build-maven-plugin-itests/tests/src/test/java/org/kie/unpackbuildplugin/integrationtests/UnpackBuildIT.java [hunk 3]
```diff
@@ -31,6 +31,7 @@ public void testBuildProjectUsingInvoker() throws IOException, VerificationExcep
 
     private void testBuildProject(String projectFolder) throws IOException, VerificationException {
         final File testDir = ResourceExtractor.simpleExtractResources(getClass(), projectFolder);
+        System.out.println("Test dir is " + testDir);
```
Comment (MarianMacik) [inline unpack-build-maven-plugin-itests/tests/src/test/java/org/kie/unpackbuildplugin/integrationtests/UnpackBuildIT.java:34]:
```suggestion
```

File: pom.xml [hunk 4]
```diff
@@ -105,52 +106,4 @@
       </dependency>
     </dependencies>
   </dependencyManagement>
-
-  <build>
-    <pluginManagement><!-- lock down plugins versions to avoid using Maven defaults (may be moved to parent pom) -->
```
Comment (MarianMacik) [inline pom.xml:110]: No longer needed?

File: README.md [hunk 5]
```diff
@@ -1,3 +1,80 @@
...
+        <goal>unpack-build</goal>
+      </goals>
+      <configuration>
+        <rootDirectory>${project.basedir}</rootDirectory>
+        <version>${unpackbuild.version}</version>
+      </configuration>
+    </execution>
+  </executions>
+</plugin>
+```
```
Comment (MarianMacik) [inline README.md:41]: Maybe also the invoker option could be added.

**Comment by baldimir** [comment]
Sorry, I am closing this as it is not needed anymore. I found a more standard solution for the unpack, so maybe this plugin is not needed anymore. See e.g. here https://github.com/kiegroup/drools/blob/main/build-parent/pom.xml#L2483

**Pull request closed without merging**